src/en/news/blog/2024/silver-bullet-rocksdb-performance/index.md

```diff
@@ -0,0 +1,120 @@
+
+1) Improved performance of RocksDB itself.
+2) Enhanced write performance of OSDs, leading to faster recovery times.
+3) Reduced downtime, resulting in a smaller amount of data needing recovery.
+
+Some graphs from server metrics to show this. Our procedure for compacting OSDs involves a staggered shutdown of OSDs. Once all OSDs are shut down, compaction is performed in parallel (`df|grep "/var/lib/ceph/osd" |awk '{print $6}' |cut -d '-' -f 2|sort -n|xargs -n 1 -P 10 -I OSD ceph-kvstore-tool bluestore-kv /var/lib/ceph/osd/ceph-OSD compact`).
+
+Debug package disk IOPS (before)
+
+![](images/storage1_OSD_compact_before_perf_package_IOPS.png)
+
+Performance package disk IOPS (after)
+
+![](images/storage1_OSD_compact_after_perf_package_IOPS.png)
+
+This is a node with SATA SSDs. A node with NVMe disks is even faster. Compaction time is between 3-5 minutes.
+
+Debug package disk IOPS (before)
+
+![](images/storage23_OSD_compact_before_perf_package_IOPS.png)
+
+Performance package disk IOPS (after)
+
+![](images/storage23_OSD_compact_after_perf_package_IOPS.png)
+
+Please note the increasing difference in OSD compaction time between SATA SSDs and DVMe drives. Previously, this gap was not as big due to the performance issues with RocksDB. However, with the introduction of faster and lower-latency disks, this difference has become more pronounced. THis suggests that the most significant perfornance improvements can be seen in clusters equipped with faster disks. 
+
+In this specific cluster, the performance packages have reduced the time needed to compact all OSDs by approximately one-third. Previously taking nine and a half hours, the process now completes in six hours.
+
+### Real world performance gains
+
+While we hoped to see the performance double this was not the case. However we still saw some significant improvements in performance. To detect [gray failure](https://www.microsoft.com/en-us/research/wp-content/uploads/2017/06/paper-1.pdf) We have virtual machines running on our cloud to continously (at 5 minute intervals) monitor performance as perceived from within the virtual machine. One of these tests is a [fio](https://github.com/axboe/fio) 4K random write test (single threaded, queue depth of 1).
+
+4K random write on CephFS (before)
+
+![](images/benchmark_VM_before_perf_package_CephFS_randwrite_IOPS.png)
+
+4K random write on CephFS (after)
+
+![](images/benchmark_VM_after_perf_package_CephFS_randwrite_IOPS.png)
+
+We have also performed other fio benchmarks and the main benefits are that standard deviation is lower, and tail latencies have decreased.
+
+
+### Conclusions
+
+- It pays off to conduct performance verification against a "known good" cluster, as Mark Nelson did. This helps identify performance issues early, before running production workloads. 
+- Clusters equipped with faster disks are more likely to see significant performance improvements compared to those with lower-performance drives like SSDs or rotational media.
```

Comment:
like SAS/SATA SSDs or HDDs.

Reply:
Thanks Anthony. I have incorporated your changes and will push a new version.